02-express-tutorial/app.js

```diff
@@ -1 +1,63 @@
-console.log('Express Tutorial')
+const express = require("express");
+const app = express();
+const { products } = require("./data");
+
+app.use(express.static("./public"));
+
+app.get("/api/v1/test", (req, res) => {
+  res.json({ message: "It worked!" });
+});
+
+app.get("/api/v1/products", (req, res) => {
+  res.json(products);
+});
+
+app.get("/api/v1/products/:productID", (req, res) => {
+  const idToFind = parseInt(req.params.productID);
+  const product = products.find((p) => p.id === idToFind);
+  if (!product) {
+    res.json({ message: "That product was not found." });
+  } else {
+    res.json(product);
+  }
+});
+
+app.get("/api/v1/query", (req, res) => {
+  let response = [];
+  const search = req.query.search;
+  const limit = req.query.limit;
+
+  if (search != "") {
+    if (search.slice(0, 4) === "nam=") {
+      response = products.filter((product) =>
+        product.name.includes(search.slice(4))
+      );
+    } else if (search.slice(0, 5) === "desc=") {
```

Comment:
Also, a pro tip for keeping your code 'clean' is to have an if check for each scenario as to not run code that is not necessary. 

So for example, you might do something like:

```js
if (desc) { code }

if (limit) { code } 

// etc. 
```

You might have to figure out how to pull that info from the `req.query`. 

Let me know if you have any follow up questions.